### PR TITLE
Mac build

### PR DESCRIPTION
**Pull request by bam241**
this allow to build.on Mac OS 

it also an autogenerated version.hpp file for the version numbers.

### CODE REVIEW

**Comment by bam241** [comment]
when compiling on Mac OS with non-standalone mode I got:
```
Undefined symbols for architecture x86_64:
"_Gopt", referenced from: [...]
```
which is solved by defining `struct program_option_struct Gopt;` which is defined original in `option.hpp` as extern.

**Comment by bam241** [comment]
@gonuke @pshriwise @ljacobson64 this should be ready to be reviewed.

File: version.hpp.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef MCNP2CAD_VERSION_H
+#define MCNP2CAD_VERSION_H
+
+#define MCNP2CAD_VERSION_MAJOR @MCNP2CAD_VERSION_MAJOR@
+#define MCNP2CAD_VERSION_MINOR @MCNP2CAD_VERSION_MINOR@
+#define MCNP2CAD_VERSION_REV @MCNP2CAD_VERSION_PATCH@
+
+#define STANDALONE_MCNP2CAD '@STANDALONE_MCNP2CAD@'
```
Comment (gonuke) [inline version.hpp.in:8]: Is there a reason for quotes here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -90,8 +85,10 @@ set(MCNP2CAD_PUB_HEADERS dataref.hpp
                          mcnp2cad.hpp
                          MCNPInput.hpp
                          options.hpp
-                         version.hpp
```
Comment (gonuke) [inline CMakeLists.txt:93]: don't we still want this to be installed even though it is autogenerated?

**Review by gonuke — CHANGES_REQUESTED** [review]
Just a couple of little questions/changes and then happy to merge

File: CMakeLists.txt (hunk 2 above)
Comment (bam241) [reply CMakeLists.txt:93]: we do at line 91

Comment (bam241) [reply CMakeLists.txt:93]: I think ?!

File: version.hpp.in (hunk 1 above)
Comment (bam241) [reply version.hpp.in:8]: I don't quite remember.... I'll check

Comment (bam241) [reply version.hpp.in:8]: I am not sure...
I'll try without.

File: CMakeLists.txt (hunk 2 above)
Comment (gonuke) [reply CMakeLists.txt:93]: I think line 91 makes it available for the build, but doesn't install it for use by downstream applications, although I'm not 100% sure.

File: CMakeLists.txt [hunk 3]
```diff
@@ -18,15 +22,6 @@ set(CMAKE_CXX_STANDARD 11)
 # iGeom assumed to have cone functionality
 add_definitions(-DHAVE_IGEOM_CONE=ON)
 
-# Adjust compiler setting for Linux using gcc version 5.0 and higher
-if (CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
-    if (NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
-      set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
```
Comment (ljacobson64) [inline CMakeLists.txt:25]: Perhaps I am out of the loop, but how come these lines were removed? I remember them being necessary when I was building on Linux.

Comment (bam241) [reply CMakeLists.txt:25]: it is still needed for Trelis 17.1.
And we now add it manually on the reps we build (MOAB, and DAGMC).

but as this need to be removed for Cubit 2020.2, we decided that we remove them and one specify them manually in the make process.
(@pshriwise I think we need to update the doc)

File: CMakeLists.txt (hunk 2 above)
Comment (bam241) [reply CMakeLists.txt:93]: yes you are probably right...

**Comment by bam241** [comment]
@gonuke I think I addressed all your comments.

(@pshriwise @gonuke this is the next PR to get reviewed/merged in the Trellis-plugin GitHub Action chain.)

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
In the absence of CI, I'll merge this for testing downstream

**Pull request merged**